hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerSelectionCriteria.java

```diff
@@ -136,6 +140,19 @@ public NavigableSet<ContainerID> getCandidateContainers(
     });
 
     containerIDSet.removeIf(this::isContainerReplicatingOrDeleting);
+
+    // remove EC containers
+    containerIDSet.removeIf(containerID -> {
+      try {
```

Comment:
Wouldn't it be cleaner to move the try-catch block from lambda to the isECContainer() method?

Reply:
I agree with @myskov.  What's more, we could merge these conditions that require a container lookup.  Something like this:

```
    // single removeIf
    containerIDSet.removeIf(this::shouldBeExcluded);
    ...
  }

  boolean shouldBeExcluded(ContainerID id) {
    ContainerInfo container;
    try {
      container = containerManager.getContainer(id);
    } catch (ContainerNotFoundException e) {
      LOG.warn(...);
      return true;
    }

    return isClosed(container) || ... || isECContainer(container);
  }
```

Reply:
Great suggestions. Added.